client-mr/mr-shaded/pom.xml

```diff
@@ -0,0 +1,149 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+~ Licensed to the Apache Software Foundation (ASF) under one or more
+~ contributor license agreements.  See the NOTICE file distributed with
+~ this work for additional information regarding copyright ownership.
+~ The ASF licenses this file to You under the Apache License, Version 2.0
+~ (the "License"); you may not use this file except in compliance with
+~ the License.  You may obtain a copy of the License at
+~
+~    http://www.apache.org/licenses/LICENSE-2.0
+~
+~ Unless required by applicable law or agreed to in writing, software
+~ distributed under the License is distributed on an "AS IS" BASIS,
+~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+~ See the License for the specific language governing permissions and
+~ limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.celeborn</groupId>
+    <artifactId>celeborn-parent_${scala.binary.version}</artifactId>
+    <version>${project.version}</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>celeborn-client-mr-shaded-${hadoop.version}</artifactId>
+  <packaging>jar</packaging>
+  <name>Celeborn Shaded Client for MapReduce</name>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.apache.celeborn</groupId>
+      <artifactId>celeborn-client-mr-${hadoop.version}</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <configuration>
+          <relocations>
+            <relocation>
+              <pattern>com.google.protobuf</pattern>
+              <shadedPattern>${shading.prefix}.com.google.protobuf</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>com.google.common</pattern>
+              <shadedPattern>${shading.prefix}.com.google.common</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>io.netty</pattern>
+              <shadedPattern>${shading.prefix}.io.netty</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>org.apache.commons</pattern>
+              <shadedPattern>${shading.prefix}.org.apache.commons</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>com.fasterxml.jackson.core</pattern>
```

Comment:
why do we need to shade the dependency `jackson` for MapReduce client?

Reply:
Spark3 has Jackson and Hadoop 3 has Jackson. I found that they have different versions so I just shade Jackson to avoid dependencies conflicts.

Reply:
but the Celeborn client doesn't need Jackson at runtime, right?

Reply:
for spark-3.4

```
$ ./build/mvn -Pspark-3.4 dependency:tree  | grep -i jackson
Using `mvn` from path: /tmp/dddd/celeborn/build/apache-maven-3.8.8/bin/mvn
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.12.7:provided
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.14.2:provided
[INFO] |  +- org.json4s:json4s-jackson_2.12:jar:3.7.0-M11:provided
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.14.2:provided
[INFO] |  +- com.fasterxml.jackson.module:jackson-module-scala_2.12:jar:2.14.2:provided
[INFO] |  |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.13.4:provided
[INFO] |     \- org.apache.parquet:parquet-jackson:jar:1.12.3:provided
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.12.7:provided
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.14.2:provided
[INFO] |  +- org.json4s:json4s-jackson_2.12:jar:3.7.0-M11:provided
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.14.2:provided
[INFO] |  +- com.fasterxml.jackson.module:jackson-module-scala_2.12:jar:2.14.2:provided
[INFO] |  |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.13.4:provided
[INFO] |     \- org.apache.parquet:parquet-jackson:jar:1.12.3:provided
[INFO] |  |  +- org.json4s:json4s-jackson_2.12:jar:3.7.0-M11:provided
[INFO] |  |  +- com.fasterxml.jackson.module:jackson-module-scala_2.12:jar:2.14.2:provided
[INFO] |  |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.13.4:provided
[INFO] |  |  \- org.apache.parquet:parquet-jackson:jar:1.12.3:provided
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.14.2:provided
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-annotations:jar:2.14.2:provided
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.14.2:provided
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.12.7:test
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.14.2:test
[INFO] |  +- org.json4s:json4s-jackson_2.12:jar:3.7.0-M11:test
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.14.2:test
[INFO] |  +- com.fasterxml.jackson.module:jackson-module-scala_2.12:jar:2.14.2:test
[INFO] |  |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.13.4:test
[INFO] |     \- org.apache.parquet:parquet-jackson:jar:1.12.3:test
```

```
$./build/sbt -Pspark-3.4 celeborn-client-spark-3/Runtime/dependencyTree | grep -i jackson
```

All of them fall within the scope of both the `test` and `provided` categories.

Reply:
> but the Celeborn client doesn't need Jackson at runtime, right?

Yes,  Celeborn client won't need Jackson at runtime.

Reply:
I'll try to remove Jackson dependencies.